### PR TITLE
Remove conditions and parameters from stack resources

### PR DESCRIPTION
**Pull request by dominikschubert**
Changes
- Stack parameters are not part of `Stack.resources` anymore
- Stack conditions are not part of `Stack.resources` anymore

### CODE REVIEW

File: localstack/services/cloudformation/engine/template_utils.py [hunk 1]
```diff
@@ -150,10 +150,20 @@ def resolve_condition(condition, conditions, parameters, mappings, stack_name):
...
-                    # return parameters[v]
+                    parameter_type: str = param["ParameterType"]
+                    parameter_value = param.get("ResolvedValue") or param.get("ParameterValue")
+
+                    if parameter_type in ["CommaDelimitedList"] or parameter_type.startswith(
+                        "List<"
+                    ):
+                        return [p.strip() for p in parameter_value.split(",")]
+                    else:
+                        return parameter_value
```
Comment (dominikschubert) [inline localstack/services/cloudformation/engine/template_utils.py:161]: This is currently a known duplication of code. Not really necessary to refactor yet though.

File: localstack/services/cloudformation/engine/entities.py [hunk 2]
```diff
@@ -244,26 +243,10 @@ def stack_name(self):
...
-                result[name] = {
-                    "Type": "Parameter",
-                    "LogicalResourceId": name,
-                    "Properties": {"Value": value},
-                }
-
-        return result
+    def resources(self):
+        """Return dict of resources"""
+        return dict(self.template_resources)
```
Comment (dominikschubert) [inline localstack/services/cloudformation/engine/entities.py:249]: :arrow_up:  core change :sunglasses:

**Review by pinzon — APPROVED** [review]

File: localstack/services/cloudformation/engine/transformers.py [hunk 3]
```diff
@@ -53,7 +58,9 @@ def _visit(obj, **_):
             if transformer_class:
                 transformer = transformer_class()
                 parameters = transform.get("Parameters") or {}
-                parameters = resolve_refs_recursively(stack_name, resources, mappings, parameters)
+                parameters = resolve_refs_recursively(
```
Comment (simonrw) [inline localstack/services/cloudformation/engine/transformers.py:61]: Not a huge fan of adding more parameters to this function. If it needs it then so be it, and it's probably worth it to remove anything that isn't a resource from the `resources` dict.

**Review by simonrw — COMMENTED** [review]
LGTM! Glad we can finally clean up the resources dictionary

File: localstack/services/cloudformation/engine/transformers.py (hunk 3 above)
Comment (dominikschubert) [reply localstack/services/cloudformation/engine/transformers.py:61]: Yeah it's more of a temporary measure anyway, even if that temporary is a span of a few months. Just helps in being more explicit in what the specific parts of the engine need in terms of parameters, resources, etc.

**Pull request merged**